### ISSUE TITLE
Extension never get to work when open header but not open source file

### ISSUE THREAD

**Issue by rhfiesta** [microsoft/vscode-cpptools#4320]
**Type: LanguageService**

**Describe the bug**
- OS and Version:
win10 1901
- VS Code Version:
1.38.1
- C/C++ Extension Version:
0.25.1
- Other extensions you installed (and if the issue persists after disabling them):
- A clear and concise description of what the bug is.

1. open vscode with a c++ project path, c/c++ extension already installed
2. open one/more existing header file, extension never work (eg. autocomplete)
3. open a source file, extension is working
4. if a project has only header files but no source file, extension never work

**Comment by michelleangela** [comment]
Can you provide:

- A copy of c_cpp_properties.json settings if you have one configured?
- Enable [logging for the language server ](https://code.visualstudio.com/docs/cpp/enable-logging-cpp#_enable-logging-for-the-language-server) to log info when opening header files and source files, and then send the logs over?
- If available, a sample of header and source files that repros the issue?

**Comment by rhfiesta** [comment]
- Default setting.

- Error log details: 

> Quick info operation failed: Error: Failed to find translation unit. Try opening a source file that includes this header, then re-open this header. 
> using Tag Parser for quick info
> 

- Sorry, the source code is not available. But it seems that the issue may be caused by the headers including some source files which cannot be open in the search path.

- May not be an issue, but I feel not friendly enough?

**Comment by michelleangela** [comment]
Can you try including the **path** to the missing files under the property setting `includePath` in c_cpp_properties.json? The values in c_cpp_properties.json are used to configure IntelliSense, such as telling IntelliSense which defines to use, or which headers/paths to use to look for files. 

To edit the c_cpp_properties.json, you can create/edit it by running the command `C/C++: Edit Configurations (UI)` or editing the file directly.

More details of c_cpp_properties.json can be found at https://code.visualstudio.com/docs/cpp/c-cpp-properties-schema-reference. 

An example of configuring IntelliSense for MSVC compiler can be found at https://code.visualstudio.com/docs/cpp/config-msvc.

**Comment by rhfiesta** [comment]
The IntelliSense is working and autocomplete is available when all missing files are included in the searchable path.

After the autocomplete is available, other header files which include unsearchable files could also be anlyzed intelligently.

Therefore, the IntelliSense can actually handle headers that include unsearchable files, but it was unavaliable at first.

**Comment by michelleangela** [comment]
> The IntelliSense is working and autocomplete is available when all missing files are included in the searchable path.

By "searchable path", do you mean adding the header path to the `includePath` in c_cpp_properties.json?

> Therefore, the IntelliSense can actually handle headers that include unsearchable files, but it was unavaliable at first.

Are the "unsearchable files" the files the ones that needed to be set under `includePath` for IntelliSense to find?

**Comment by rhfiesta** [comment]
I found it has nothing to do with the `includePath`.

I built a header and source file sample that can reproduce the issue as follows:

ac_test.h
```
#ifndef _B_H_
#define _B_H_

struct ac_test
{
    int     test_int1;
    int     test_int2;
    int     test_int3;
    int     test_int4;
    int     test_int5;
    
    inline ac_test()
    {
        ;
    }
};

#endif
```
ac_test.cpp
```
#if 0

#include "ac_test.h"

#endif
```

1. Put `ac_test.h` and `ac_test.cpp` in a same path and open the folder in vscode.

1. Close vscode and delete all buffers (globalStorage / snippets / workspaceStorage) under `C:\Users\%Users%\AppData\Roaming\Code\User` to clear vscode env.

1. Open vscode, IntelliSense is avaliable on `ac_test.h`, autocomplete can provide suggestions for `ac_test()`.

1. Close and re-open vscode, this time IntelliSense is unavaliable on `ac_test.h`. The autocomplete does not work in `ac_test()`.

1. Reopening vscode afterwards does not make IntelliSense available.

I found the **key** that caused the issue is `#include "ac_test.h"` with `#if 0`

**Comment by sean-mcmanus** [comment]
Yeah, the problem is that we're associating headers with TUs in which the header is not actually used due to being in an inactive region.

**Comment by michelleangela** [comment]
Fixed in version 0.26.2-insiders, available at https://github.com/microsoft/vscode-cpptools/releases/tag/0.26.2-insiders.
